lib/bn/fqp.ex

```diff
@@ -0,0 +1,308 @@
+defmodule BN.FQP do
+  defstruct [:coef, :modulus_coef, :dim]
+
+  alias BN.FQ
+
+  @type t :: %__MODULE__{
+          coef: [FQ.t()],
+          modulus_coef: [integer()]
+        }
+
+  @spec new([integer()], [integer()], keyword()) :: t()
```

Comment:
I think the return type is not accurate since this function raises errors. You might have to do something like `:: t() | no_return`